src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -1381,7 +1220,9 @@ bool ObjectSynchronizer::request_deflate_idle_monitors() {
     }
     if (self->is_Java_thread()) {
       // JavaThread has to honor the blocking protocol.
-      ThreadBlockInVM tbivm(self->as_Java_thread());
+      {
+        ThreadBlockInVM tbivm(self->as_Java_thread());
+      }
       os::naked_short_sleep(999);  // sleep for almost 1 second
```

Comment:
Hi Dan, you need to be blocked while sleeping, otherwise you are blocking safepoints for "almost 1 second".
So previous code was correct.

Reply:
Good catch! Yes, I should not have written that change so late at
night. I know better. I have reverted it.